src/rosenfeld_fmt.cc

```diff
@@ -254,32 +235,499 @@ void RosenfeldFMT::_compute(std::shared_ptr<State> state, bool compute_value)
+                dphi_dnv2k_wv2k(idx) = -dphi_dnv2k(idx)*wv2;
+                // TODO: there is a missing term for nv1 and nv2
+                dphi_dnv2k_w3k(idx) = -dphi_dnv2k(idx)*w3;
+                dphi_dnv1k_w3k(idx) = -dphi_dnv1k(idx)*w3;
+                }
+
+            ft_->setReciprocalData(dphi_dn0k_w0k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dn0->full_view().begin());
+
+            ft_->setReciprocalData(dphi_dn1k_w1k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dn1->full_view().begin());
+
+            ft_->setReciprocalData(dphi_dn2k_w2k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dn2->full_view().begin());
+
+            ft_->setReciprocalData(dphi_dn3k_w3k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dn3->full_view().begin());
+
+            ft_->setReciprocalData(dphi_dnv1k_wv1k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dnv1->full_view().begin());
+            
+            ft_->setReciprocalData(dphi_dnv2k_wv2k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dnv2->full_view().begin());
+            
+            ft_->setReciprocalData(dphi_dnv2k_w3k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dnv2_w3->full_view().begin());
+            
+            ft_->setReciprocalData(dphi_dnv1k_w3k);
+            ft_->transform();
+            std::copy(ft_->const_view_real().begin(),ft_->const_view_real().end(),tmp_dF_dnv1_w3->full_view().begin());
+            }
+            
+        
+        // normalize phi derivatives with r
+            {        
+            auto dF_dn0 = tmp_dF_dn0->view();
+            auto dF_dn1 = tmp_dF_dn1->view();
+            auto dF_dn2 = tmp_dF_dn2->view();
+            auto dF_dn3 = tmp_dF_dn3->view();
+            auto dF_dnv1 = tmp_dF_dnv1->view();
+            auto dF_dnv2 = tmp_dF_dnv2->view();
+            auto dF_dnv2_w3 = tmp_dF_dnv2_w3->view();
+            auto dF_dnv1_w3 = tmp_dF_dnv1_w3->view();
+            
+            for(int idx = 0; idx < mesh->shape(); ++idx)
+                {           
+                const auto r = mesh->center(idx);
+                dF_dn0(idx) /= r;
+                dF_dn1(idx) /= r;
+                dF_dn2(idx) /= r;
+                dF_dn3(idx) /= r;
+                dF_dnv1(idx) /= r;
+                dF_dnv2(idx) /= r;
+                // TODO: extra term for nv1 and nv2
+                dF_dnv2(idx) = dF_dnv2_w3(idx)/(r*r)+dF_dnv2(idx);
+                dF_dnv1(idx) = dF_dnv1_w3(idx)/(r*r)+dF_dnv1(idx);
```

Comment:
Let's add a comment and use `+=` so it's obvious what this is doing
```suggestion
                // consolidate second term with first
                dF_dnv2(idx) += dF_dnv2_w3(idx)/(r*r);
                dF_dnv1(idx) += dF_dnv1_w3(idx)/(r*r);
```